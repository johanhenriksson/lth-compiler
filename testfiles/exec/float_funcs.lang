int main() {
    float i = promote(5);
    i = i + 1.25; // i = 6.25

    float f = 0.56;
    return truncate(f) + truncate(i) - 6; // returns 0
}
