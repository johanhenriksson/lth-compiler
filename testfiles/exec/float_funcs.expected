; Library Functions
declare i8* @__malloc(i32)
declare void @__delete(i8*)
declare void @print(i64)
declare i64 @read()
declare i64 @truncate(double)
declare double @promote(i64)
declare void @assert(i1)
declare void @except(i1)

define i64 @main() {
%i = alloca double
%_1 = call double @promote(i64 5)
store double %_1, double* %i
; IdUse i
%_3 = load double, double* %i
%_2 = fadd double %_3, 1.25
store double %_2, double* %i
%f = alloca double
store double 0.56, double* %f
; IdUse f
%_6 = load double, double* %f
%_7 = call i64 @truncate(double %_6)
; IdUse i
%_8 = load double, double* %i
%_9 = call i64 @truncate(double %_8)
%_5 = add i64 %_7, %_9
%_4 = sub i64 %_5, 6
ret i64 %_4
}

